chainer/training/extensions/log_report.py

```
@@ -0,0 +1,115 @@
+    time that the trigger fires.
+
+    It also adds ``'epoch'`` and ``'iteration'`` entries to each result
+    dictionary, which are the epoch and iteration counts at the output.
+
+    Args:
+        keys (iterable of strs): Keys of values to accumulate. If this is None,
+            all the values are accumulated and output to the log file.
+        trigger: Trigger that decides when to aggregate the result and output
+            the values. This is distinct from the trigger of this extension
+            itself. If it is a tuple in the form ``<int>, 'epoch'`` or
+            ``<int>, 'iteration'``, it is passed to :class:`IntervalTrigger`.
+        postprocess: Callback to postprocess the result dictionaries. Each
+            result dictionary is passed to this callback on the output. This
+            callback can modify the result dictionaries, which are used to
+            output to the log file.
+        log_name (str): Name of the log file under the output directory. It can
+            be a format string: the last result dictionary is passed for the
+            formatting. For example, users can use '{.iteration}' to separate
+            the log files for different iterations. If the log name is None, it
+            does not output the log to any file.
+
+    """
+    def __init__(self, keys=None, trigger=(1, 'epoch'), postprocess=None,
+                 log_name='log'):
+        self._keys = keys
+        self._trigger = trigger_module.get_trigger(trigger)
+        self._postprocess = postprocess
+        self._log_name = log_name
+        self._log = []
+
+        self._init_summary()
+
+    def __call__(self, trainer):
+        # accumulate the observations
+        keys = self._keys
+        observation = trainer.observation
+        summary = self._summary
+
+        if keys is None:
+            summary.add(observation)
+        else:
+            summary.add({k: observation[k] for k in keys if k in observation})
```

It looks like this will ignore any invalid keys (not contained in observation). Is this okay? Should it raise an exception? Should the behavior for invalid keys be documented in the initializer?